programs/bpf/rust/invoke/src/lib.rs

```diff
@@ -263,10 +263,9 @@ fn process_instruction(
                     )?,
                     accounts[DERIVED_KEY1_INDEX].key
                 );
-                let not_native_program_id = Pubkey::new_from_array([6u8; 32]);
-                assert!(!not_native_program_id.is_native_program_id());
+                let new_program_id = Pubkey::new_from_array([6u8; 32]);
```

Comment:
nit: Could switch these to `Pubkey::new_unique()` while we're here

Reply:
Yeah, I'd like to keep this pr clean, can do a mas s// to new_unique in a separate pr